scripts/run_yaml_cde_calculation_tests.py

```diff
@@ -0,0 +1,140 @@
+    parser.add_argument("test_csvs", metavar="CSV", nargs="+",
+                        type=argparse.FileType("r"),
+                        help="CSV files containing test definitions")
+    parser.add_argument("--outfile", metavar="FILE", type=argparse.FileType("w"),
+                        help="Output file")
+    args = parser.parse_args()
+
+    if not args.outfile: args.outfile = sys.stdout
+
+    registry = load_yaml(args.registry_yaml[0])
+
+    success = True
+    for infile in args.test_csvs:
+        success = run_tests(registry, infile, args.outfile) and success
+    return 0 if success else 1
+
+Registry = namedtuple("Registry", ("names", "calculations"))
+
+def load_yaml(file_obj):
+    calculations = {}
+    names = {}
+    data = yaml.load(file_obj)
+    for cde in data.get("cdes") or []:
+        if cde.get("code"):
+            calc = cde.get("calculation")
+            if calc:
+                calculations[cde["code"]] = calc
+        names[cde["code"]] = cde.get("name")
+    return Registry(names, calculations)
+
+TestResult = namedtuple("TestResult", ("test", "expected", "actual", "error", "output"))
+TestCase = namedtuple("TestCase", ("file", "number", "check_code", "params"))
+
+def run_tests(registry, csv_file, outfile):
+    num_tests = 0
+    num_success = 0
+    num_errors = 0
+    for index, row in enumerate(csv.DictReader(csv_file)):
+        t = setup_test(row, index + 1, csv_file.name)
+        num_tests += 1
+        res = run_test(registry, t)
+
+        if res.error:
+            print_error(registry, res, row, outfile)
+            num_errors += 1
+        elif res.expected == res.actual:
+            num_success += 1
+        else:
+            print_failure(registry, res, row, outfile)
+
+    return num_tests == num_success
+
+def setup_test(cols, num, filename):
+    check_code = cols["check"]
+    params = ((code, val) for code, val in cols.items() if code != "check")
+    return TestCase(filename, num, check_code, dict(params))
+
+def load_adsafe_js():
+    adsafe_path = "../rdrf/rdrf/static/js/vendor/adsafe-min.js"
+    return io.open(os.path.join(os.path.dirname(__file__), adsafe_path)).read()
+
+def run_test(registry, test):
+    context = dict((code, val) for code, val in test.params.items() if code in registry.names)
+    patient = dict((code, val) for code, val in test.params.items() if code not in registry.names)
+    script = registry.calculations[test.check_code]
+
+    script = u"""
+        "use strict";
+        var document = {};
+        %s
+
+        var patient = %s;
+        var context = %s;
+        var RDRF = ADSAFE;
+
+        %s
+
+        console.log(context.result);
+    """ % (load_adsafe_js(), json.dumps(patient), json.dumps(context), script)
+
+    success, output = exec_script(script)
+
+    if success:
+        lines = [o.strip() for o in output.split("\n")]
+        non_empty = filter(bool, lines)
+        context_result = non_empty[-1] if non_empty else None
```

Comment:
My only concern would be we're testing  the result of writing context.result to a log file not context.result directly - am I right?